django/middleware/common.py

```diff
@@ -113,7 +113,8 @@ def process_response(self, request, response):
             if self.should_redirect_with_slash(request):
                 return self.response_redirect_class(self.get_full_path_with_slash(request))
 
-        if settings.USE_ETAGS:
+        need_etag = 'no-store' not in response.get('Cache-Control', '')
```

Comment:
This needs to be more robust, since `Cache-Control` is a comma-separated field and can contain arbitrary [extensions](https://tools.ietf.org/html/rfc7234#section-5.2.3). That is, it's perfectly valid for someone to use `Cache-Control: my-directive="my-no-store"`, which your test would mistakenly match.

Also, values should be compared case-insensitively.

See [`has_vary_header()`](https://github.com/django/django/blob/ac77c55bc5fc54cd763a7ae426784650a8cc97c9/django/utils/cache.py#L254) for the basic approach. Trying to do this in a general way (e.g. creating `django.utils.cache.has_cache_control_header()`) would be more complicated since the `Cache-Control` values have an optional argument, but we probably don't need a general solution to this problem if it's not used elsewhere.